scripts/core/kernel.yml

```diff
@@ -534,3 +534,47 @@ params:
 returns:
     - $X_RESULT_ERROR_UNSUPPORTED_FEATURE:
         - "If the adapter has no underlying equivalent handle."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the suggested local work-item number from runtime implementation."
+class: $xKernel
+name: GetSuggestedLocalWorkSize
+ordinal: "0"
+details:
+    - "pLocalWorkSize can be omitted in urEnqueueKernelLaunch(), but beside from"
+    - "OpenCL, LocalWorkSize will need to be calculated or guessed before enqueue"
+    - "the kernel. This function will get the LocalWorkSize value used when enqueueing"
+    - "the kernel."
```

Comment:
```suggestion
    - "Query a suggested group size for a kernel given a global size for each dimension."
    - "The application may call this function from simultaneous threads for the same context."
    - "The implementation of this function should be thread-safe."
```

Reply:
Since `urEnqueueKernelLaunch` does not guarantee thread-safe and most of the API's implementation comes from `urEnqueueKernelLaunch`, we will need to do some careful examination to see if this API's implementation could be thread-safe.